### DISCUSSION TITLE
Removing a nested array of empty object

### DISCUSSION THREAD

**Discussion by mmahalwy** [react-hook-form/react-hook-form#5952] [Q&A]
**Describe the bug**
When using a nested array of empty objects, it is removed from values when edit is made. 

**To Reproduce**
1. Go to https://codesandbox.io/s/awesome-herschel-rifm6?file=/src/App.js
2. Notice the values
3. Edit the input on the right panel
4. Notice that the values no longer has the empty `initialValues: [{}]`

Oddly, this only happens with `initialValues` is NOT the only key. If you remove `hello` key in the example provided, you'll notice it keeps `initialValues` in the values 🤷‍♂️ 

**Codesandbox link (Required)**
https://codesandbox.io/s/awesome-herschel-rifm6?file=/src/App.js

**Expected behavior**
Expectation is that the values maintain the array of `initialValues`

**Comment by mmahalwy** [comment]
After clicking around different versions of react-hook-form, seems like this bug was introduced at 7.8.4. All previous versions do not have this bug.

**Comment by bluebill1049** [comment]
<img width="910" alt="Screen Shot 2021-07-15 at 4 16 52 pm" src="https://user-images.githubusercontent.com/10513364/125738299-3cc10c7f-5e6d-4c68-a9cd-1cdfefb0f546.png">

you have only register a single input 

`<input {...register("title")} />`

so react hook form will register those defaultValues on your behalf, when you start to type into the input, the value will be retrieved from the input itself and hence the deeply nested input gets removed. 

## work around

- register those inputs which you have declared in the defaultValues

**Reply by mmahalwy** [reply]
@bluebill1049 I think that's overkill for dynamic forms. Can't predict all the keys to register them beforehand. Lastly, this does not explain why it works when you:
Have other keys in the same object
```
{ initialValues: [{}], hello: 1}
```
Or there are keys on initialValues
```
{ initialValues: [{ hi: 2 }], hello: 1}
```
Feels like this is a regression.

Lastly, in the example you provided, what happens to the `test1` array? How do you register it?!

**Reply by bluebill1049** [reply]
have a read at this RFC, hopefully will give you more context: https://github.com/react-hook-form/react-hook-form/discussions/5087

**Reply by mmahalwy** [reply]
@bluebill1049 read it. still doesnt explain why in your example `test1` is removed in the submit data. If anything, should still be there according to RFC :)
